doc/exts/pylint_options.py

```diff
@@ -0,0 +1,190 @@
+    """Registers all checkers and extensions found in the default folders."""
+    initialize_checkers(linter)
+    initialize_extensions(linter)
+
+
+def _get_all_options(linter: PyLinter) -> OptionsDataDict:
+    """Get all options registered to a linter and return the data."""
+    all_options: OptionsDataDict = defaultdict(list)
+    for checker in sorted(
+        linter.get_checkers(),
+        key=lambda x: (
+            x.name != "master",
+            getmodule(x).__name__.startswith("pylint.extensions."),  # type: ignore[union-attr]
+            x.name,
+        ),
+    ):
+        ch_name = checker.name
+        for option in checker.options:
+            all_options[ch_name].append(
+                OptionsData(
+                    option[0],
+                    option[1],
+                    checker,
+                    getmodule(checker).__name__.startswith("pylint.extensions."),  # type: ignore[union-attr]
+                )
+            )
+
+    return all_options
+
+
+def _create_checker_section(
+    checker: str, options: list[OptionsData], linter: PyLinter
+) -> str:
+    checker_string = get_rst_title(f"``{checker.capitalize()}`` Checker", "^")
+
+    toml_doc = tomlkit.document()
+    pylint_tool_table = tomlkit.table(is_super_table=True)
+    toml_doc.add(tomlkit.key(["tool", "pylint"]), pylint_tool_table)
+
+    checker_table = tomlkit.table()
+
+    for option in sorted(options, key=lambda x: x):
+        checker_string += get_rst_title(f"--{option.name}", '"')
+        checker_string += f"\nDescription: *{option.optdict.get('help')}*\n\n"
+        checker_string += f"Default: *{option.optdict.get('default')}*\n\n\n"
+
+        # Start adding the option to the toml example
+        if option.optdict.get("hide_from_config_file"):
+            continue
+
+        # Get current value of option
+        value = getattr(linter.config, option.name.replace("-", "_"))
+
+        # Create a comment if the option has no value
+        if value is None:
+            checker_table.add(tomlkit.comment(f"{option.name} ="))
+            checker_table.add(tomlkit.nl())
+            continue
+
+        # Tomlkit doesn't support regular expressions
+        if isinstance(value, re.Pattern):
+            value = value.pattern
+        elif (
+            isinstance(value, (list, tuple))
+            and value
+            and isinstance(value[0], re.Pattern)
+        ):
+            value = [i.pattern for i in value]
+
+        # Add to table
+        checker_table.add(option.name, value)
+        checker_table.add(tomlkit.nl())
+
+    pylint_tool_table.add(options[0].checker.name.lower(), checker_table)
+    toml_string = "\n".join(f"   {i}" for i in tomlkit.dumps(toml_doc).split("\n"))
+    checker_string += f"""
+.. raw:: html
+
+   <details>
+   <summary><a>Example configuration section</a></summary>
+
+**Note:** Only ``pylint.tool`` is required, the section title is not. These are the default values.
+
+.. code::
```

Comment:
Maybe use `.. code:: toml` for toml syntax highlighting. If none I will use python syntax highlighting
if none:
![image](https://user-images.githubusercontent.com/55311527/163671108-6a231625-2f67-4cca-a273-ed38f5a733f3.png)
if use `.. code::toml`:
![image](https://user-images.githubusercontent.com/55311527/163671130-d32b7629-fdae-41c6-8e36-be93d51d98d9.png)

Reply:
Oh I think I got a warning previously that that directive wasn't recognise. Let me recheck!

Reply:
`.. code:: toml`
`.. code:: toml:` not valid I think
edit: already fix 👍

Reply:
Yes, thanks for the tip!